### DISCUSSION TITLE
How to properly modify attribute of `Batch` objects?

### DISCUSSION THREAD

**Discussion by ekosman** [pyg-team/pytorch_geometric#6298] [Q&A] [answered]
Assuming I have already constructed a `Batch` object, that could contain either `Data` or `HeteroData` - what is the proper way to modify its attributes? modifying them directly won't work because the objects returned from `Batch.to_data_list()` won't recognize all attributes. For example:

```
b = Batch.from_data_list([HeteroData()])
b["a"].x = torch.randn(3,3)
data = b.to_data_list()[0]
contains_attrs = "x" in data["a"]
print(f"Does the data contains any nodes? {contains_attrs }")
```

Edit: I just noticed that all attributes must be contained within `_slice_dict` - which is created only if `from_data_list` was used to create the batch. Is there a pre implemented way to manually add items to this dictionary?

**Comment by rusty1s** [comment, marked as answer]
The only way currently would be to modify `_slice_dict` by yourself for this. Otherwise, this is not possible as we don't know how to split `x` in this case :(
